src/main/java/core/basesyntax/service/impl/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionService;
+import core.basesyntax.service.ReaderService;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    private static final String TITLE = "type";
+    private static final String SEPARATOR = ",";
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final int OPERATION_INDEX = 0;
+    private static final int TRANSACTION_DATA_QUANTITY = 3;
+
+    private final ReaderService readerService = new ReaderServiceImpl();

```

Comment:
Consider using Dependency Injection for the ReaderService instead of directly initializing it. This will make the code more modular and easier to test.

Reply:
I haven't studied Dependency Injection so I don't know how to use it properly. Please don't take this as a mistake.

Reply:
It meant that it is better not to initialize directly in the class. It is better to transfer all internal dependencies from the outside.